app/code/VConnect/OrderVolume/Model/Order/Item/OrderItemsVolumeCalculator.php

```diff
@@ -0,0 +1,95 @@
+<?php
+declare(strict_types=1);
+
+namespace VConnect\OrderVolume\Model\Order\Item;
+
+use Magento\Catalog\Model\Product;
+use Magento\Sales\Api\Data\OrderItemInterface;
+use VConnect\OrderVolume\Model\Product\Config as ProductConfig;
+
+class OrderItemsVolumeCalculator
+{
+    private array $orderItemsVolumeData = [];
+
+    public function calculate(array $orderItems): void
+    {
+        /** @var OrderItemInterface $orderItem */
+        foreach ($orderItems as $orderItem) {
+            $this->setCalculationResult((int)$orderItem->getItemId(), $this->getOrderItemVolume($orderItem));
+        }
+    }
+
+    /**
+     * @param OrderItemInterface $orderItem
+     * @return float|int
+     * @throws \Exception
+     */
+    private function getOrderItemVolume(OrderItemInterface $orderItem): float|int
+    {
+        $orderItemQuantity = $orderItem->getQtyOrdered();
+        $productVolume = $this->getProductVolume($orderItem);
+        if ($this->isFloat($productVolume)) {
```

Comment:
what is the reason of this? You can float integer variables and it will work fine as well.

https://onlinephp.io/c/5f259